### PR TITLE
Better scheduling

### PR DESCRIPTION
**Pull request by shans**
* support for fixed tasks that cut time out from other tasks
* fix a few scheduling bugs
* allow fixed tasks to apply to multiple people

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ async function main() {
     tabs = tabsToData(data);
   }
 
-  const fullTabs = layoutGantt(tabs);
+  const fullTabs = improvedLayout(tabs);
```
Comment (shaper) [inline src/main.ts:32]: Do we still need `layoutGantt` imported here? It does look like we still call it from `index.ts` and `byOwner.ts`. Should we move those to use the improved one eventually?

**Review by shaper — APPROVED** [review]
I joked about unit tests but they would actually be useful to be able to see what bugs/edge cases there were and of course gain confidence for future refactoring that we don't re-introduce them. Just commenting since it's clear there's well more than enough logic for them to provide benefit.

File: src/main.ts (hunk 1 above)
Comment (shans) [reply src/main.ts:32]: Fixed

**Comment by shans** [comment]
I've added some light unit tests, probably not enough yet to show up edge cases but it's a start.
